src/common/HPMDataCheck.h

```diff
@@ -326,7 +326,7 @@ HPExport const struct s_HPMDataCheck HPMDataCheck[] = {
 		#define COMMON_HPMI_H
 	#endif // COMMON_HPMI_H
 	#ifdef COMMON_MAPCHARPACKETS_H
-		{ "PACKET_MAPCHAR_AGENCY_JOIN_PARTY_REQ", sizeof(struct PACKET_MAPCHAR_AGENCY_JOIN_PARTY_REQ), SERVER_TYPE_ALL },
+		{ "PACKET_MAPCHAR_AUTH_REQ", sizeof(struct PACKET_MAPCHAR_AUTH_REQ), SERVER_TYPE_ALL },
```

Comment:
Changed to draft to avoid accidental merge;;

This seems wrong... I did not remove this structure and I am not sure why HPMHookGen is generating that... 

If I move the declaration to between `PACKET_MAPCHAR_AGENCY_JOIN_PARTY_REQ` and `PACKET_MAPCHAR_GUILD_EMBLEM` in mapcharpackets.h, then `PACKET_MAPCHAR_GUILD_EMBLEM` gets removed.

any ideas?

Reply:
Did you already try deleting `tools/HPMHookGen/doxyoutput/`? When I get weird output it's usually due to stale data in there and doxygen not regenerating it

Reply:
yes, I did (found about it today, ever wondered how this worked 😅 ) but deleting it and running `make` again still gives this output

Reply:
It seems that the macro call `DEFINE_PACKET_ID()` without a trailing semicolon is throwing doxygen off and making it ignore the following struct definition.

The same is happening in `charloginpackets.h` to `PACKET_CHARLOGIN_ONLINE_ACCOUNTS` and in `charmappackets.h` to `PACKET_CHARMAP_GUILD_INFO_EMPTY` and `PACKET_CHARMAP_GUILD_INFO_EMBLEM`.

Adding semicolons at the end of macro calls seems a quick and dirty way to prevent doxygen from getting confused.
The actual root issue though is that these `*packets.h` headers that use the `DEFINE_PACKET_ID()` macro forgot to include `common/packetsmacro.h` (and some of them probably only include it indirectly and by coincidence). Adding that include explicitly solves the issue, allowing doxygen to replace the macros and parse the file correctly. I don't see any ill side-effects in adding it everywhere it's needed.

Reply:
Ahh nice!! It works perfectly. I added an extra commit updating other files that calls `DEFINE_PACKET_ID` and did not import `packetsmacro`, there were a few other entries missing